docs/arch/adr-16.md

```diff
@@ -0,0 +1,40 @@
+# ADR 16: Monorepo Tooling
+
+Created: July 3, 2019
+
+## Context
+
+When we first started the rebuild, @rogerhutchings set it up as a monorepo. The idea being that there are benefits to separating our code between multiple apps and packages:
+
+  - We can reuse code more easily
+  - Error boundaries are built-in (if we break one thing, we don't necessarily break everything)
+  - To make changes to one thing, we don't need to recompile and rebuild _everything_ in development, at least
+
+...but that working across multiple repos can be pretty tedious: adding a feature which requires changes to more than one package requires multiple pull requests to different repos, for example. So Roger set it up as a monorepo using Lerna. This started out fine, but we hit some issues in the development of Planet Hunters TESS:
+
+  - `styled-components` wasn't collect styling from symlinked dependencies, causing a FOUC on initial render: https://github.com/zooniverse/front-end-monorepo/issues/327
+  - Adding the `@zooniverse/standard` package broke the build because Lerna wasn't correctly linking it up: https://github.com/zooniverse/front-end-monorepo/pull/958
```

Comment:
Something that's been puzzling me: why does the linter break a production build?

Reply:
The production build script was doing a standard `npm install` without the `--production` flag. I _had_ added the `NODE_ENV=production` envinronment variable, but it seemed to be ignored (possibly as the install command was being run in a subshell? not bash-savvy enough, but I thought they inherited their environment). I started noticing the broken linking locally too, but transiently 🤦‍♂ see https://github.com/zooniverse/front-end-monorepo/issues/953

Reply:
Also, the way the build process is set up, we'd have to install the build dependencies for the `lib-` folders, as we need to build a production version of the libraries for the apps to use.